pkg/federation/sdlmerge/collect_valid_entities.go

```diff
@@ -0,0 +1,121 @@
+package sdlmerge
+
+import (
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astvisitor"
+	"github.com/jensneuse/graphql-go-tools/pkg/operationreport"
+)
+
+type collectValidEntitiesVisitor struct {
```

Comment:
Code is a bit hard to understand it is worth adding comments on how it works or try to simplify the logic